src/ripple/core/TimeKeeper.h

```diff
@@ -22,73 +22,99 @@
 
 #include <ripple/basics/chrono.h>
 #include <ripple/beast/clock/abstract_clock.h>
-#include <ripple/beast/utility/Journal.h>
-#include <string>
-#include <vector>
+#include <atomic>
 
 namespace ripple {
 
 /** Manages various times used by the server. */
 class TimeKeeper : public beast::abstract_clock<NetClock>
 {
-public:
-    virtual ~TimeKeeper() = default;
-
-    /** Launch the internal thread.
+private:
+    std::atomic<std::chrono::seconds> closeOffset_{};
 
-        The internal thread synchronizes local network time
-        using the provided list of SNTP servers.
-    */
-    virtual void
-    run(std::vector<std::string> const& servers) = 0;
-
-    /** Returns the estimate of wall time, in network time.
+    // Adjust system_clock::time_point for NetClock epoch
+    static constexpr time_point
+    adjust(std::chrono::system_clock::time_point when)
+    {
+        return time_point(std::chrono::duration_cast<duration>(
+            when.time_since_epoch() - days(10957)));
```

Comment:
Just a nit: we usually like to separate the implementation from the interface so it's easier to see interfaces. I'm OK merging this as-is though. It's only a nit and we can clean it up latter (although I did want to flag it; for future changes we should try to stick with separate interfaces/implementations).